tiledb/sm/subarray/range_subset.cc

```diff
@@ -131,6 +132,15 @@ RangeSetAndSuperset::RangeSetAndSuperset(
     ranges_.emplace_back(superset);
 }
 
+RangeSetAndSuperset::RangeSetAndSuperset(
+    Datatype datatype,
+    const Range& superset,
+    std::vector<Range> subset,
```

Comment:
Is there any reason this can't be std::vector<Range>&& subset? That way we don't create a copy of the ranges on the constructor. We can also do : ranges_(subset)